plugin/src/test/java/org/opensearch/ml/rest/SecureMLRestIT.java

```diff
@@ -133,6 +138,70 @@ public void testTrainAndPredictWithFullMLAccessNoIndexAccess() throws IOExceptio
         );
     }
 
+    public void testUploadModelWithNoAccess() throws IOException {
+        exceptionRule.expect(ResponseException.class);
+        exceptionRule.expectMessage("no permissions for [cluster:admin/opensearch/ml/upload_model]");
+        uploadModel(mlNoAccessClient, TestHelper.toJsonString(mlUploadInput), null);
+    }
+
+    public void testUploadModelWithReadOnlyMLAccess() throws IOException {
+        exceptionRule.expect(ResponseException.class);
+        exceptionRule.expectMessage("no permissions for [cluster:admin/opensearch/ml/upload_model]");
+        uploadModel(mlReadOnlyClient, TestHelper.toJsonString(mlUploadInput), null);
+    }
+
+    public void testUploadModelWithFullMLAccessNoIndexAccess() throws IOException {
+        exceptionRule.expect(ResponseException.class);
+        exceptionRule.expectMessage("no permissions for [cluster:admin/opensearch/ml/upload_model]");
+        uploadModel(mlFullAccessNoIndexAccessClient, TestHelper.toJsonString(mlUploadInput), null);
+    }
+
+    public void testUploadModelWithFullAccess() throws IOException {
+        uploadModel(mlFullAccessClient, TestHelper.toJsonString(mlUploadInput), uploadModelResult -> {
+            assertFalse(uploadModelResult.containsKey("model_id"));
+            String taskId = (String) uploadModelResult.get("task_id");
+            assertNotNull(taskId);
+            String status = (String) uploadModelResult.get("status");
+            assertEquals(MLTaskState.CREATED.name(), status);
+            try {
+                getTask(mlFullAccessClient, taskId, task -> {
+                    String algorithm = (String) task.get("function_name");
+                    assertEquals(FunctionName.TEXT_EMBEDDING.name(), algorithm);
+                });
+            } catch (IOException e) {
+                assertNull(e);
+            }
+        });
+    }
+
+    public void testLoadModelWithNoAccess() throws IOException {
+        exceptionRule.expect(RuntimeException.class);
+        exceptionRule.expectMessage("no permissions for [cluster:admin/opensearch/ml/load_model]");
+        loadModel(mlNoAccessClient, mlUploadInput, null);
+    }
+
+    public void testLoadModelWithReadOnlyMLAccess() throws IOException {
+        exceptionRule.expect(ResponseException.class);
+        exceptionRule.expectMessage("no permissions for [cluster:admin/opensearch/ml/load_model]");
+        loadModel(mlReadOnlyClient, mlUploadInput, null);
+    }
+
+    public void testLoadModelWithFullMLAccessNoIndexAccess() throws IOException {
+        exceptionRule.expect(ResponseException.class);
+        exceptionRule.expectMessage("no permissions for [cluster:admin/opensearch/ml/load_model]");
+        loadModel(mlFullAccessNoIndexAccessClient, mlUploadInput, null);
+    }
+
+    public void testLoadModelWithFullAccess() throws IOException {
```

Comment:
Also add test cases for new unload and profile APIs?

Reply:
Yes, let's add them in the next PR